README.rst

```diff
@@ -128,7 +128,7 @@ Here is the code:
             return produce_expensive_data()
 
         # get the temp directory shared by all workers
-        root_tmp_dir = tmp_path_factory.getbasetemp().parent
+        root_tmp_dir = tmp_path_factory.getbasetemp()

```

Comment:
as per https://github.com/pytest-dev/pytest-xdist/blob/10ec3af8a04e61348e0cadba9e36078be9ac5783/src/xdist/workermanage.py#L257-L258 on a worker the shared tmpdir of the manager is used

Reply:
Oh, I see my mistake now. My confusion happened because I'm always reading/writing the data from/to the file (even when running with a single worker or without xdist). I didn't realize the basetemp could change it for workers.

Thanks for pointing this out and sorry for the noise.